src/PartitionLoops.cpp

```diff
@@ -723,21 +723,21 @@ class RenormalizeGPULoops : public IRMutator {
             return;
         }
 
-        if (ends_with(op->name, "__thread_id_x")) {
-            in_thread_loop = true;
-            IRMutator::visit(op);
-            in_thread_loop = false;
-            return;
-        }
-
         bool old_in_gpu_loop = in_gpu_loop;
 
         if (in_gpu_loop || CodeGen_GPU_Dev::is_gpu_var(op->name)) {
             gpu_vars.push(op->name, 0);
             in_gpu_loop = true;
         }
 
-        IRMutator::visit(op);
+
+        if (ends_with(op->name, "__thread_id_x")) {
+            in_thread_loop = true;
+            IRMutator::visit(op);
+            in_thread_loop = false;
```

Comment:
If we aren't going to save/restore the previous value, I'd like to assert that it is false before setting it to true above.